gallery/src/ThemeSettingsModal/ThemeSettingsModal.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, useState } from 'react';
+
+import {
+  NxButton,
+  NxModal,
+  NxFormGroup,
+  NxToggle,
+  NxFieldset,
+  NxRadio,
+  NxTooltip,
+  NxFontAwesomeIcon,
+  NxCode,
+  useToggle
+} from '@sonatype/react-shared-components';
+
+import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';
+
+import { themingEnabled, themeOverride, setThemingEnabled, setThemeOverride } from '../themeClassManager';
+
+type Props = {
+  onClose: () => void;
+};
+
+const ThemeSettingsModal = ({ onClose }: Props) => {
+  const [themingEnabledState, toggleThemingEnabledState] = useToggle(themingEnabled),
+      [themeOverrideState, setThemeOverrideState] = useState(themeOverride);
+
+  useEffect(function() {
+    setThemingEnabled(themingEnabledState);
+  }, [themingEnabledState]);
+
+  useEffect(function() {
+    setThemeOverride(themeOverrideState);
+  }, [themeOverrideState]);
+
+  return (
+    <NxModal id="nx-modal-dark-mode-example"
```

Comment:
I'm not sure if this id is needed.

Reply:
> I'm not sure if this id is needed.

My thought was to have an id to make it easier for testing as an easy selector. However, I can remove it if you prefer!

Reply:
> > I'm not sure if this id is needed.
> 
> My thought was to have an id to make it easier for testing as an easy and specific selector. However, I can remove it if you prefer!

Oo hmm, I wonder if we could remove this id and add a classname instead, maybe something like `gallery-theme-settings-modal`

Reply:
> > > I'm not sure if this id is needed.
> > 
> > 
> > My thought was to have an id to make it easier for testing as an easy and specific selector. However, I can remove it if you prefer!
> 
> Oo hmm, I wonder if we could remove this id and add a classname instead, maybe something like `gallery-theme-settings-modal`

Changed here: [af68c48](https://github.com/sonatype/sonatype-react-shared-components/pull/782/commits/af68c4892214e891dcd143b2b0fa1fc4c71c04e2)